src/python/WMCore/Algorithms/BasicAlgos.py

```diff
@@ -10,7 +10,48 @@
 import os
 import stat
 import time
-import hashlib
+import zlib
+import subprocess
+
+def calculateChecksums(filename):
+    """
+    _calculateChecksums_
+
+    Get the adler32 and crc32 checksums of a file. Return None on error
+
+    Process line by line and adjust for known signed vs. unsigned issues
+      http://docs.python.org/library/zlib.html
+
+    The cksum UNIX command line tool implements a CRC32 checksum that is
+    different than any of the python algorithms, therefore open cksum
+    in a subprocess and feed it the same chunks of data that are used
+    to calculate the adler32 checksum.
+
+    """
+    adler32Checksum = 1 # adler32 of an empty string
+    cksumProcess = subprocess.Popen("cksum", stdin = subprocess.PIPE, stdout = subprocess.PIPE)
+
+    f = open(filename, 'rb')
```

Comment:
Just read up on it. No functional difference and the old style syntax isn't deprecated or anything like that either. There are no benefits for a simple example like this here, although I can see it being useful if what you ar doing in the file processing section is more complex.